### PR TITLE
Made version number a module variable.

### PR DESCRIPTION
**Pull request by latture**
This PR moves the version number to `src/citrine/__version__.py`, so it can be shared with user docs. This pattern was modeled after a similar solution used by the `requests` package.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/__version__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__version__ = '0.85.3'
```
Comment (latture) [inline src/citrine/__version__.py:1]: Link to equivalent file in `requests`: https://github.com/psf/requests/blob/master/requests/__version__.py

They put more information in that file that just the version number. We could do something similar in the future, but this provides a pattern we could expand on in the future.

File: setup.py [hunk 2]
```diff
@@ -60,9 +60,13 @@ def run(self):
         strip_all_hints(STRIP_DIRS)
         build_py.run(self)
 
+about = {}
+with open(join(this_directory, 'src', 'citrine', '__version__.py'), 'r', 'utf-8') as f:
+    exec(f.read(), about)
```
Comment (latture) [inline setup.py:65]: Very similar pattern to https://github.com/psf/requests/blob/master/setup.py#L60

File: docs/source/conf.py [hunk 3]
```diff
@@ -14,16 +14,19 @@
...
 copyright = '2019, Citrine Informatics'
 author = 'Citrine Informatics'
 
-# The full version, including alpha/beta/rc tags
-release = '0.0.1'
-
+# The short X.Y version.
+version = citrine.__version__
+# The full version, including alpha/beta/rc tags.
+release = citrine.__version__
```
Comment (latture) [inline docs/source/conf.py:29]: https://github.com/psf/requests/blob/master/docs/conf.py#L68

File: setup.py [hunk 4]
```diff
@@ -60,9 +60,13 @@ def run(self):
         strip_all_hints(STRIP_DIRS)
         build_py.run(self)
 
+about = {}
+with open(join(this_directory, 'src', 'citrine', '__version__.py'), 'r') as f:
+    exec(f.read(), about)
```
Comment (latture) [inline setup.py:65]: Very similar pattern to https://github.com/psf/requests/blob/master/setup.py#L60

**Comment by latture** [comment]
@maxhutch This isn't something we have to do, but this is one way we could provide a shared version number.

File: src/citrine/__init__.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 # TODO: Add a docstring here
 from citrine.citrine import Citrine  # noqa: F401
 import logging
+from .__version__ import __version__
```
Comment (talltea) [inline src/citrine/__init__.py:4]: Build failures are coming from this import being unused

**Review by talltea — COMMENTED** [review]
This looks good to me. I think we also need to change the script that checks version bumps (https://github.com/CitrineInformatics/citrine-python/blob/master/scripts/validate-version-bump.sh) to point to `src/citrine/__version__.py`

File: src/citrine/__init__.py (hunk 5 above)
Comment (latture) [reply src/citrine/__init__.py:4]: 👍

**Comment by latture** [comment]
> This looks good to me. I think we also need to change the script that checks version bumps (https://github.com/CitrineInformatics/citrine-python/blob/master/scripts/validate-version-bump.sh) to point to `src/citrine/__version__.py`

You're right. (As someone who's terrible at writing regular expressions) do you know what that regex should be?

**Review by talltea — DISMISSED** [review]
This looks good to me 👍  (as long as we can merge in the follow up pr changing the version bump script to point to the correct file in master)

**Review by maxhutch — DISMISSED** [review]
LGTM

**Comment by latture** [comment]
This needed a version bump. Re-requesting review. Also, the follow up PR is queued up here: #496

**Review by maxhutch — APPROVED** [review]

**Pull request merged**